### PR TITLE
Create a table to log when firmware moves between different targets

### PR DESCRIPTION
**Pull request by hughsie**
I'm not super happy with the "Promotions" name -- any better ideas welcome. It's really a table that hold what user moved the firmware to different targets, and when they did it.

### CODE REVIEW

**Comment by hughsie** [comment]
screenshot from 2018-02-23 20-59-12

**Comment by hughsie** [comment]
Ideas while sitting on a bus: Note, FirmwareNote, FirmwareEntry, FirmwareEvent, FirmwareRecord, FirmwareStatement, FirmwareNote, FirmwareAction or FirmwareStatus.

**Comment by superm1** [comment]
For a table name `channel_movements`?

File: app/db.py [hunk 1]
```diff
@@ -35,13 +35,68 @@ def init_app(self, app):
...
+                    spl = e.message.split(' ')
+                    if len(spl) != 5:
+                        continue
+                    filename = spl[2]
+                    if filename not in fws_filename:
+                        print('filename %s does not exist, skipping' % filename)
+                        continue
+                    if e.username not in users:
+                        if e.username == 'dell':
+                            user_id = users['dellqa'].user_id
```
Comment (superm1) [inline app/db.py]: Is dell a special case I guess?

File: app/models.py [hunk 2]
```diff
@@ -360,6 +361,31 @@ def find_req(self, kind=None, value=None):
     def __repr__(self):
         return "Component object %s" % self.appstream_id
 
+class Promotion(db.Base):
```
Comment (superm1) [inline app/models.py]: If you're not crazy about `Promotion`, how about `Channel`?

File: app/db.py (hunk 1 above)
Comment (hughsie) [reply app/db.py]: Yup, when we started we had no user/qa split. Just Dell that's affected as it was the first real user.

Comment (superm1) [reply app/db.py]: Probably worth leaving a comment in the code around that.

Comment (hughsie) [reply app/db.py]: This is the hacky db migration code; it'll be gone in a few commits after being deployed. :)

Comment (superm1) [reply app/db.py]: OK :)

**Pull request merged**